packages/headless/src/features/commerce/search/search-actions.ts

```diff
@@ -165,7 +165,7 @@ export const prepareForSearchWithQuery = createAsyncThunk<
       query: payload.query,
     })
   );
-  dispatch(updatePage(1));
+  dispatch(selectPage({page: 0}));
```

Comment:
I also noticed we weren't triggering the proper page selection action. Now we do.